packages/app/src/systems/Transaction/utils/icon.ts

```diff
@@ -1,19 +1,15 @@
-import { Icon } from '@fuel-ui/react';
+import type { Icons } from '@fuel-ui/react';
 
 import { OperationName } from './tx.types';
 
-export const getTxIcon = (type?: OperationName) => {
-  switch (type) {
-    case OperationName.transfer:
-      return Icon.is('UploadSimple');
-    case OperationName.receive:
-      return Icon.is('DownloadSimple');
-    case OperationName.mint:
-    case OperationName.predicatecall:
-      return Icon.is('MagicWand');
-    case OperationName.contractCall:
-      return Icon.is('ArrowsLeftRight');
-    default:
-      return 'ArrowRight';
-  }
+const ICON_MAP = {
+  [OperationName.transfer]: 'UploadSimple',
+  [OperationName.receive]: 'DownloadSimple',
+  [OperationName.mint]: 'ArrowRight',
+  [OperationName.predicatecall]: 'MagicWand',
+  [OperationName.contractCall]: 'ArrowsLeftRight',
+};
+
+export const getTxIcon = (type?: OperationName): Icons => {
+  return type ? ICON_MAP[type] ?? 'ArrowRight' : 'ArrowRight';
```

Comment:
wouldn't this code have the same effect?

`return ICON_MAP[type] || 'ArrowRight'`

Reply:
It would but typescript requires a check as  `type` can be `undefined`, and there's not `undefined` key in `ICON_MAP`.

<img width="636" alt="Screenshot 2023-01-17 at 3 39 37 AM" src="https://user-images.githubusercontent.com/15849997/212804788-e48eebe9-19c4-45ed-bcd7-1140aa24084c.png">